<launch>
    <!-- Select a object shape: "rectangular_prism, triangular_prism, polygonal_prism" -->
    <arg name="object_shape" default="rectangular_prism" />
    <param name="object_shape" value="$(arg object_shape)" />

    <!-- Rectangular Prism -->
    <group if="$(eval object_shape == 'rectangular_prism')">
        <rosparam command="load" file="$(find grf_transport)/config/rectangular_prism.yaml" />
        <node name="spawn_urdf_box" pkg="gazebo_ros" type="spawn_model"
            args="-urdf -model $(arg object_shape) -x 0.0 -y 1.1 -z 0.23 -file $(find grf_transport)/models/rectangular_prism/rectangular_prism.urdf" />

        <node name="spawn_urdf_box_target" pkg="gazebo_ros" type="spawn_model"
            args="-urdf -model target_$(arg object_shape)  -x 0.0 -y -1.2 -z 0.22 -file $(find grf_transport)/models/rectangular_prism/rectangular_prism_target.urdf" />
    </group>

    <!-- Triangular Prism -->
    <group if="$(eval object_shape == 'triangular_prism')">
        <rosparam command="load" file="$(find grf_transport)/config/triangular_prism.yaml" />
        <node name="spawn_urdf_box" pkg="gazebo_ros" type="spawn_model"
            args="-urdf -model $(arg object_shape) -x 0.0 -y 1.1 -z 0.23 -file $(find grf_transport)/models/triangular_prism/triangular_prism.urdf" />

        <node name="spawn_urdf_box_target" pkg="gazebo_ros" type="spawn_model"
            args="-urdf -model target_$(arg object_shape)  -x -0.0 -y -1.2 -z 0.22 -file $(find grf_transport)/models/triangular_prism/triangular_prism_target.urdf" />
    </group>
    
    <!-- Polygonal Prism -->
    <group if="$(eval object_shape == 'polygonal_prism')">
        <rosparam command="load" file="$(find grf_transport)/config/polygonal_prism.yaml" />
        <node name="spawn_urdf_box" pkg="gazebo_ros" type="spawn_model"
            args="-urdf -model $(arg object_shape) -x 0.0 -y 1.1 -z 0.23 -file $(find grf_transport)/models/polygonal_prism/polygonal_prism.urdf" />

        <node name="spawn_urdf_box_target" pkg="gazebo_ros" type="spawn_model"
            args="-urdf -model target_$(arg object_shape)  -x 0.0 -y -1.2 -z 0.22 -file $(find grf_transport)/models/polygonal_prism/polygonal_prism_target.urdf" />
    </group>

</launch>